challenge/files/11_maps.toml

```diff
@@ -0,0 +1,189 @@
+Which of the following statements best describes how these buckets work?
+
+"""
+type = "select_answers"
+answers = [
+    { text = "Each bucket can store exactly one key-value pair" },
+    { text = "Buckets use a linked list to store multiple key-value pairs that hash to the same bucket" },
+    { text = "Buckets store key-value pairs in an array and resolve collisions using open addressing" },
+    { text = "Buckets store key-value pairs in an array and use a linked list to handle collisions", code_line_ranges = [] },
+]
+
+[[questions]]
+text = """What happens when the number of entries in a Go map exceeds a certain threshold?
+
+"""
+type = "select_answers"
+answers = [
+    { text = "The map allocates a new array with double the capacity and rehashes all keys" },
+    { text = "The map remains the same size but uses a more sophisticated hash function" },
+    { text = "The map splits existing buckets and rehashes only a portion of the keys", code_line_ranges = [] },
+    { text = "The map creates a second, independent map and balances the keys between the two maps" },
+]
+
+[[questions]]
+text = """The load factor of a map is a critical concept in understanding its efficiency. 
+Which of the following best describes the load factor in the context of Go maps?
+
+"""
+type = "select_answers"
+answers = [
+    { text = "The ratio of the number of entries to the total capacity of the map" },
+    { text = "The ratio of the number of used buckets to the total number of buckets" },
+    { text = "The average number of key-value pairs stored in each bucket", code_line_ranges = [] },
+    { text = "The ratio of the number of overflow buckets to the total number of buckets" },
+]
+
+
+[[questions]]
+text = """How does the Go runtime handle memory management and garbage collection for maps, particularly regarding unused keys?
+
+"""
+type = "select_answers"
+answers = [
+    { text = "The Go runtime immediately reclaims memory for a map entry when a key is deleted" },
+    { text = "The Go runtime marks deleted keys as unused, and the memory is reclaimed during the next garbage collection cycle", code_line_ranges = [] },
+    { text = "The Go runtime never reclaims memory for deleted keys; the memory remains allocated until the entire map is garbage collected" },
+    { text = "The Go runtime reclaims memory for deleted keys only when the map is resized" },
+]
+
+[[questions]]
+text = """Which of the following statements about the hash function used in Go maps is true?
+
+"""
+type = "select_answers"
+answers = [
+    { text = "Go uses a simple modulo operation to compute hash values for keys" },
+    { text = "The hash function is deterministic and always produces the same hash for the same input across different runs" },
+    { text = "The hash function includes a per-map random seed to protect against certain types of attacks", code_line_ranges = [] },
+    { text = "Go maps do not use a hash function; they rely on direct indexing" },
+]
+
+
```

Comment:
We need to add the question "What happens when we need to add new value, but the bucket is full"